docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,340 @@
+``gammapy analysis config`` command line tool which dumps the config file with all lines commented
+out, and the users can then uncomment and fill in the parameters and values they care about. As an 
+alternative users could copy & paste from a config file example in the docs. We will develop a 
+schema and validate / give good error messages on read.
+
+Prototype configuration file.
+
+.. code::    
+    
+    analysis:
+        process:
+            # add options to allow either in-memory or disk-based processing
+            out_folder: "."  # default is current working directory
+            store_per_obs: {true, false}
+        reduce:
+            type: {"1D, "3D"}
+            stacked: {true, false}
+            background: {"irf", "reflected", "ring"}
+            roi: max_offset
+            exclusion: exclusion.fits
+        fit:
+            energy_min, energy_max
+        logging:
+            level: debug
+
+    grid:
+        spatial: center, width, binsz
+        energy: min, max, nbins
+        time: min, max
+        # PSF RAD and EDISP MIGRA not exposed for now
+        # Per-obs energy_safe and roi_max ?
+    
+    observations:
+        data_store: $GAMMAPY_DATA/cta-1dc/index/gps/
+        ids: 110380, 111140, 111159
+        conesearch: 
+            ra:
+            dec:
+            radius:
+            energy_min:
+            energy_max:
+            time_min:
+            time_max:
+
+    model:
+        # Model configuration will mostly be designed in different PIG
+        sources:
+            source_1: spectrum: powerlaw, spatial: shell
+            diffuse: gal_diffuse.fits
+        background:
+            IRF
+
+**API design**
+
+The design of the high-level interface API is driven by the use cases considered, the different
+tasks (tools) identified and their responsibilities, as well as the need of a main `Analysis` 
+session object that drives and orchestrates the different tools involved, their inputs and products. 
+The `Analysis` session object will be initialized with a configuration file (see *prototype 
+configuration file* above) and will be the responsible to instatiate and run the different tools 
+classes. The tools are middle  management agents (e.g. MapMaker, ReflectedBgEstimator, ...) 
+responsible to perform the different  tasks identified in the use cases covered. `Analysis` needs
+to know how to turn configuration values into Python objects (e.g. geometries, exclusion_mask: 
+exclusion.fits to make a Map object) and also how to instatiate and run the different tools. A 
+possible solution for **dispatching** among `Analysis` and tools may be the one proposed below.
+
+.. code::
+
+    class Analysis:
+        def reduce_data(self):
+            config = self.make_data_reduction_config(self.config)
+            maker = self.make_data_recution_class(config)
+            maker.run()
+            self.datasets = maker.datasets
+
+We will need "registries" mapping options to tools classes, e.g.:
```

Comment:
Suggest to simplify and remove this part.
Don't show any example how the flexible toolchain might look.

Instead say here: we will get this high-level API to work, but if we can create an extensible framework for the underlying tools and toolchains is unclear. In any case, end-users will get the high-level API described here.